src/Api/Serializer/UserSerializer.php

```diff
@@ -19,14 +19,14 @@ protected function getDefaultAttributes($user)
     {
         $attributes = parent::getDefaultAttributes($user);
 
-        $canEdit = $this->actor->can('edit', $user);
-
         $attributes += [
-            'joinTime'         => $this->formatDate($user->joined_at),
-            'discussionCount'  => (int) $user->discussion_count,
-            'commentCount'     => (int) $user->comment_count,
-            'canEdit'          => $canEdit,
-            'canDelete'        => $this->actor->can('delete', $user),
+            'joinTime'           => $this->formatDate($user->joined_at),
+            'discussionCount'    => (int) $user->discussion_count,
+            'commentCount'       => (int) $user->comment_count,
+            'canEditCredentials' => $this->actor->can('user.editCredentials'),
+            'canEditGroups'      => $this->actor->can('user.editGroups'),
```

Comment:
These should be applied on the $user object so policies can work

Reply:
Also, I don't think we need the `user` prefix

Reply:
The user prefix is required, in my testing without the user. prefix it did not work correctly and always returned false. This is one of those cases of the magic not working correctly.

Reply:
The "magic" only applies when it's run through User Policy, which happens when the subject argument is a user. The prefix needs to be removed at the same time that the second user argument is added.